### ISSUE TITLE
JIT error after running for a long time

### ISSUE THREAD

**Issue by nikhilmishra000** [pytorch/pytorch#22170]
## 🐛 Bug

I'm getting a cryptic error from a JIT model that I cannot reproduce consistently:
* I'm calling the model in a loop about once per second
* The error consistently occurs after 1~2 hours of doing this
* As far as I can tell, it is unrelated to the values of the model inputs.

This is the stack trace -- any idea what this means / how to investigate?
```
RuntimeError: strides[cur-1] == sizes[cur]*strides[cur] ASSERT FAILED at /pytorch/torch/csrc/jit/fuser/executor.cpp:149, please report a bug to PyTorch. (compressContiguous at /pytorch/torch/csrc/jit/fuser/executor.cpp:149)
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7f73072b2021 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7f73072b18ea in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0x7c691e (0x7f728d41d91e in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #3: <unknown function> + 0x7c6b59 (0x7f728d41db59 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #4: torch::jit::fuser::launchFusion(torch::jit::fuser::FusedKernel const&, c10::Device, c10::ArrayRef<at::Tensor> const&, std::vector<at::Tensor, std::allocator<at::Tensor> >&) + 0xa00 (0x7f728d41e570 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #5: torch::jit::fuser::runFusion(long, std::vector<c10::IValue, std::allocator<c10::IValue> >&) + 0xcf3 (0x7f728d41faa3 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #6: torch::jit::runFusion(long, std::vector<c10::IValue, std::allocator<c10::IValue> >&) + 0x11 (0x7f728d3417d1 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #7: <unknown function> + 0x6f5cf8 (0x7f728d34ccf8 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #8: <unknown function> + 0x672286 (0x7f728d2c9286 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #9: torch::jit::InterpreterState::run(std::vector<c10::IValue, std::allocator<c10::IValue> >&) + 0x22 (0x7f728d2c4842 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #10: <unknown function> + 0x65c6ac (0x7f728d2b36ac in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #11: <unknown function> + 0x3c7a74 (0x7f72ce158a74 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
frame #12: <unknown function> + 0x3ac263 (0x7f72ce13d263 in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
frame #13: <unknown function> + 0x11642e (0x7f72cdea742e in /opt/conda/envs/lab42/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>
```

## Environment

 - PyTorch Version (e.g., 1.0):  1.0.1
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6
 - CUDA/cuDNN version: cuda 10.0, cudnn 7.4
 - GPU models and configuration: 2080ti, nvidia driver 410
 - Any other relevant information:

**Comment by dementrock** [comment]
I am also observing this issue. Help appreciated!

**Comment by jamesr66a** [comment]
@ngimel can you take a look? Seems related to fusion on the GPU

**Comment by ngimel** [comment]
From the assert message it looks like actual input tensors contiguity (the number of dimensions and the dimensions that can be flattened because they occupy contiguous physical memory regions) and contiguity expected by compiled kernel don't match. Kernel is compiled/retrieved from cache here https://github.com/pytorch/pytorch/blob/v1.0.1/torch/csrc/jit/fuser/executor.cpp#L319-L330, so it looks like created arg_spec somehow does not reflect true tensor contiguity. 
It is not related to fusion on GPU per se, this is device-agnostic code. It would happen on GPU only because CPU fusion is (mostly) disabled. 
Without repro it is hard to say what exactly is going wrong. 
Can you please try with pytorch 1.1? A lot has been changed in the argspec creation between 1.0.1 and 1.1, so it could be fixed?
If not, can we get a repro script?
If your inputs are always the same size/contiguity (e.g. always 4d contiguous tensors) then this theory is wrong, because same kernel should always be retrieved from cache, and error would either appear or not appear on the first iteration.

**Comment by nikhilmishra000** [comment]
Seems to be resolved on version 1.1 -- thanks for the help!
